homeassistant/components/meater/sensor.py

```diff
@@ -10,10 +21,108 @@
     CoordinatorEntity,
     DataUpdateCoordinator,
 )
+from homeassistant.util import dt as dt_util
 
 from .const import DOMAIN
 
 
+@dataclass
+class MeaterSensorEntityDescription(SensorEntityDescription):
+    """Describes meater sensor entity."""
+
+    available: Callable = lambda x: NotImplementedError
+    value: Callable = lambda x: NotImplementedError
+
+
+def _elapsed_time_to_timestamp(probe: MeaterProbe):
+    """Convert elapsed time to timestamp."""
+    if not probe.cook:
+        return None
+    return dt_util.utcnow() - timedelta(seconds=probe.cook.time_elapsed)
+
+
+def _remaining_time_to_timestamp(probe: MeaterProbe):
+    """Convert remaining time to timestamp."""
+    if not probe.cook or probe.cook.time_remaining < 0:
+        return None
+    return dt_util.utcnow() + timedelta(probe.cook.time_remaining)
+
+
+SENSOR_TYPES: dict[str, MeaterSensorEntityDescription] = {
```

Comment:
This doesn't have to be a dictionary? The key isn't looked up, so instead, we could use a tuple (and save a tiny bit of memory)